Improve documentation and README

nemi-notrace opened the pull request.

Frando commented on REPCO-SPEC.md:
```diff
@@ -1,4 +1,4 @@
-# repco
+# repco [no longer up to date]
```
Let's add a paragraph below the header like

*This specification is currently in review and will be updated soon. Please check the [doc](doc) folder for up-to-date descriptions of the data model.*

or something like that.

Frando commented:
Let's also rename the files. I propose something like:

```
docs/datamodel.md
docs/datasources.md
docs/repo.md
docs/packages.md
```

or similar

nemi-notrace commented:
> Let's maybe also rename the files into a simpler structure? I propose something like:
> 
> ```
> docs/datamodel.md
> docs/datasources.md
> docs/repo.md
> docs/packages.md
> ```
> 
> or similar

Changed name and folder struct, also added an extra file for the RLDM

Frando commented:
I'm going ahead and merging this. It's much better to what we had and we can refine in subsequent PRs.

Frando commented:
Can you remove the non-markdown changes from the PR? Let's keep this limited to the markdown docs I'd say. Or resolve the conflicts.

Frando commented:
I merged this in manually. Closing.
